src/js/components/Calendar/__tests__/Calendar-test.js

```diff
@@ -149,4 +164,297 @@ describe('Calendar', () => {
+          onSelect={onSelect}
+          range
+          animate={false}
+        />
+      </Grommet>,
+    );
+    fireEvent.click(getByText('17'));
+    fireEvent.click(getByText('20'));
+    expect(onSelect).toBeCalledWith([
+      [
+        expect.stringMatching(/^2020-07-17T/),
+        expect.stringMatching(/^2020-07-20T/),
+      ],
+    ]);
+  });
+
+  test('select date greater and less than', () => {
+    const onSelect = jest.fn();
+    const { getByLabelText } = render(
+      <Grommet>
+        <Calendar
+          dates={[['2018-01-01T00:00:00-08:00', '2018-01-05T00:00:00-08:00']]}
+          onSelect={onSelect}
+          range
+          animate={false}
+        />
+      </Grommet>,
+    );
+    // select date greater than January 1st
+    fireEvent.click(getByLabelText('Wed Jan 03 2018'));
+    expect(onSelect).toBeCalledWith([
+      [
+        expect.stringMatching(/^2018-01-03T/),
+        expect.stringMatching(/^2018-01-05T/),
+      ],
+    ]);
+    // select date less than January 3rd
+    fireEvent.click(getByLabelText('Mon Jan 01 2018'));
+    expect(onSelect).toBeCalledWith([
+      [
+        expect.stringMatching(/^2018-01-01T/),
+        expect.stringMatching(/^2018-01-05T/),
+      ],
+    ]);
+  });
+
+  test('select date with same start date', () => {
+    const onSelect = jest.fn();
+    const { getByLabelText } = render(
+      <Grommet>
+        <Calendar
+          dates={[['2018-01-01T00:00:00-08:00', '2018-01-03T00:00:00-08:00']]}
+          onSelect={onSelect}
+          range
+          animate={false}
+        />
+      </Grommet>,
+    );
+    // selecting same starting day
+    fireEvent.click(getByLabelText('Mon Jan 01 2018'));
+    expect(onSelect).toBeCalledWith(expect.stringMatching(/^2018-01-03T/));
+  });
+
+  test('select date with same date twice', () => {
+    const onSelect = jest.fn();
+    const { getByLabelText } = render(
+      <Grommet>
+        <Calendar
+          reference="2018-01-01T00:00:00-08:00"
+          onSelect={onSelect}
+          range
+          animate={false}
+        />
+      </Grommet>,
+    );
+    fireEvent.click(getByLabelText('Wed Jan 03 2018'));
+    expect(onSelect).toHaveBeenNthCalledWith(
+      1,
+      expect.stringMatching(/^2018-01-03T/),
+    );
+    fireEvent.click(getByLabelText('Wed Jan 03 2018'));
+    expect(onSelect).toHaveBeenNthCalledWith(2, undefined);
+  });
+
+  test('select date with same end date', () => {
+    const onSelect = jest.fn();
+    const { getByLabelText } = render(
+      <Grommet>
+        <Calendar
+          dates={[['2018-01-01T00:00:00-08:00', '2018-01-03T00:00:00-08:00']]}
+          onSelect={onSelect}
+          range
+          animate={false}
+        />
+      </Grommet>,
+    );
+    // selecting same ending day
+    fireEvent.click(getByLabelText('Wed Jan 03 2018'));
+    expect(onSelect).toBeCalledWith(expect.stringMatching(/^2018-01-01T/));
+  });
+
+  test('undefined dates', () => {
```

Comment:
How will that behave? Is it consistent enough for testing?

Reply:
I am going to remove this test because it is not consistent enough for testing. When the date is undefined the calendar is setup to display the current month so when next month comes around and this test is run the snapshot will fail.

Reply:
Good catch.